osu.Game/Overlays/Mods/ModColumn.cs

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Humanizer;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Utils;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+#nullable enable
+
+namespace osu.Game.Overlays.Mods
+{
+    public class ModColumn : CompositeDrawable
+    {
+        private Func<Mod, bool>? filter;
+
+        public Func<Mod, bool>? Filter
+        {
+            get => filter;
+            set
+            {
+                filter = value;
+                updateFilter();
+            }
+        }
+
+        private readonly ModType modType;
```

Comment:
This really needs to be renamed to `ModCategory` I think...

Reply:
Do you mean just the field or the entire enum with it? Because if it's the latter then I'm likely going to do it in a separate PR and make this one depend on it.

Reply:
Yeah, unrelated to this PR. Just a passing remark.

Reply:
Going to leave this as a follow up item either way pending final decision if you're OK with that, I don't want to complicate this branch's history too much if it can be avoided.